Docs: Updated Reporting docs (#99665)
grafana/grafana: docs/sources/dashboards/create-reports/_index.md

description: Generate and share PDF reports from your Grafana dashboards
weight: 600
refs:

>   change-ui-theme:

Author: The diff is confusing here. I added, removed, and moved refs into alphabetical order.

Any changes you make to a dashboard used in a report are reflected the next time the report is sent.

## Requirements

For Grafana Enterprise, using the Reporting feature has the following requirements:

- SMTP must be configured for reports to be sent. Refer to [SMTP configuration documentation](ref:smtp) for more information.
- The [Grafana image renderer plugin](/grafana/plugins/grafana-image-renderer) (v3.0+) must be installed or the remote rendering service must be set up. Refer to [Image rendering](ref:image-rendering) for more information.

### Rendering configuration

By default, attachments (PDFs, CSV files, and embedded images) larger than 10 MB are not sent, which keeps email servers from rejecting the email. You can increase or decrease this limit in the [reporting configuration](ref:reporting-configuration).

When a report file is generated, it's temporarily written to the corresponding folder (`csv`, `pdf`, `png`) in the Grafana `data` folder.
A background job runs every 10 minutes and removes temporary files.
You can configure how long a file should be stored before being removed by configuring the [temp_data_lifetime](ref:temp-data-lifetime) setting.

> These options are available in the [reporting configuration](ref:reporting-configuration) of the `ini` file for Enterprise Grafana.

Author: ```suggestion
```

## Access control

Only organization administrators can create reports by default.
You can customize who can create reports with [role-based access control (RBAC)](ref:rbac).

When [RBAC](ref:rbac) is enabled, you need to have the relevant [permissions](ref:permission) to create and manage reports.
Refer to specific guides to understand what permissions are required.

## Create a report

> The report creation process is multi-step, but you don't need to complete these steps in order and you can skip steps by clicking a step name at the top of the page:

Author: ```suggestion
The report creation process is multi-step, but you don't need to complete these steps in order and you can skip steps by clicking a step name at the top of the page. You can also save a draft of the report at any point during the creation process:
```

Any changes you make to a dashboard used in a report are reflected the next time the report is sent.

## Requirements

For Grafana Enterprise, using the Reporting feature has the following requirements:

- SMTP must be configured for reports to be sent. Refer to [SMTP configuration documentation](ref:smtp) for more information.
- The [Grafana image renderer plugin](/grafana/plugins/grafana-image-renderer) (v3.0+) must be installed or the remote rendering service must be set up. Refer to [Image rendering](ref:image-rendering) for more information.

### Rendering configuration

By default, attachments (PDFs, CSV files, and embedded images) larger than 10 MB are not sent, which keeps email servers from rejecting the email. You can increase or decrease this limit in the [reporting configuration](ref:reporting-configuration).

These options are available in the [reporting configuration](ref:reporting-configuration) of the `ini` file for Enterprise Grafana.

When a report file is generated, it's temporarily written to the corresponding folder (`csv`, `pdf`, `png`) in the Grafana `data` folder.
A background job runs every 10 minutes and removes temporary files.

> You can configure how long a file should be stored before being removed by configuring the [temp_data_lifetime](ref:temp-data-lifetime) setting.

Author: ```suggestion
You can configure how long a file should be stored before being removed by configuring the [`temp_data_lifetime`](ref:temp-data-lifetime) setting.
```

      destination: /docs/grafana/<GRAFANA_VERSION>/setup-grafana/configure-grafana/#temp_data_lifetime
    - pattern: /docs/grafana-cloud/
      destination: /docs/grafana/<GRAFANA_VERSION>/setup-grafana/configure-grafana/#temp_data_lifetime
  templates-and-variables:
    - pattern: /docs/grafana/
      destination: /docs/grafana/<GRAFANA_VERSION>/dashboards/variables/
    - pattern: /docs/grafana-cloud/
      destination: /docs/grafana-cloud/visualizations/dashboards/variables/
  time-range-controls:
    - pattern: /docs/grafana/
      destination: /docs/grafana/<GRAFANA_VERSION>/dashboards/use-dashboards/#set-dashboard-time-range
    - pattern: /docs/grafana-cloud/
      destination: /docs/grafana-cloud/visualizations/dashboards/use-dashboards/#set-dashboard-time-range
---

# Create and manage reports

**Reporting** allows you to send automated and scheduled emails from any of your dashboards.
You can configure several elements of these reports and generate PDFs, CSV files, and embedded images.
Any changes you make to a dashboard used in a report are reflected the next time the report is sent.

## Requirements

> - SMTP must be configured for reports to be sent. Refer to [SMTP configuration documentation](ref:smtp) for more information.

Author: ```suggestion
For Grafana Enterprise, using the Reporting feature has the following requirements:

- SMTP must be configured for reports to be sent. Refer to [SMTP configuration documentation](ref:smtp) for more information.
```

- The [Grafana image renderer plugin](/grafana/plugins/grafana-image-renderer) (v3.0+) must be installed or the remote rendering service must be set up. Refer to [Image rendering](ref:image-rendering) for more information.

### Rendering configuration

When generating reports, each panel renders separately before being collected in a PDF.
You can configure the per-panel rendering timeout and number of concurrently rendered panels.

> To make a panel more legible, you can set a scale factor for the rendered images.
> However, a higher scale factor increases the file size of the generated PDF.

Author: ```suggestion
```

You can also specify custom fonts that support different Unicode scripts.
The DejaVu font is the default used for PDF rendering.

> By default, attachments (PDFs, CSV files, and embedded images) larger than 10Mb are not sent, which keeps email servers from rejecting the email. You can increase or decrease this limit in the [reporting configuration](ref:reporting-configuration).

Reviewer: By default, attachments (PDFs, CSV files, and embedded images) larger than 10 MB are not sent, which keeps email servers from rejecting the email

> By default, attachments (PDFs, CSV files, and embedded images) larger than 10Mb are not sent, which keeps email servers from rejecting the email. You can increase or decrease this limit in the [reporting configuration](ref:reporting-configuration).

Author: ```suggestion
By default, attachments (PDFs, CSV files, and embedded images) larger than 10 MB are not sent, which keeps email servers from rejecting the email. You can increase or decrease this limit in the [reporting configuration](ref:reporting-configuration).
```

      destination: /docs/grafana/<GRAFANA_VERSION>/dashboards/variables/
    - pattern: /docs/grafana-cloud/
      destination: /docs/grafana-cloud/visualizations/dashboards/variables/
  time-range-controls:
    - pattern: /docs/grafana/
      destination: /docs/grafana/<GRAFANA_VERSION>/dashboards/use-dashboards/#set-dashboard-time-range
    - pattern: /docs/grafana-cloud/
      destination: /docs/grafana-cloud/visualizations/dashboards/use-dashboards/#set-dashboard-time-range
---

# Create and manage reports

**Reporting** allows you to send automated and scheduled emails from any of your dashboards.
You can configure several elements of these reports and generate PDFs, CSV files, and embedded images.
Any changes you make to a dashboard used in a report are reflected the next time the report is sent.

## Requirements

- SMTP must be configured for reports to be sent. Refer to [SMTP configuration documentation](ref:smtp) for more information.
- The [Grafana image renderer plugin](/grafana/plugins/grafana-image-renderer) (v3.0+) must be installed or the remote rendering service must be set up. Refer to [Image rendering](ref:image-rendering) for more information.

> 

Author: Do these requirements for Enterprise only? Can we clarify what the requirements are for Enterprise vs Cloud?

Author: Okay I'm going to add that in just to make it super clear.

> <!-- are these requirements for Enterprise only? Can we clarify what the requirements are for Enterprise vs Cloud? -->

Author: ```suggestion
```

### Rendering configuration

When generating reports, each panel renders separately before being collected in a PDF.
You can configure the per-panel rendering timeout and number of concurrently rendered panels.

To make a panel more legible, you can set a scale factor for the rendered images.

> However, a higher scale factor increases the file size of the generated PDF.

Author: Does line above apply to any file type or just PDF?

> 
> <!-- does above apply to any file type? -->

Author: ```suggestion
```

1. When you've finished making changes, click **Confirm** at the top of the screen to go to the last step.
1. Click **Update report**.

### Pause or resume reports {#pause-a-report}

You can pause and resume sending reports from the report list view.
To do this, follow these steps:

1. In the main menu, click **Dashboards > Reporting**.
1. On the row of the report you want to update, do one of the following:

   - Click the pause icon - The report won't be sent according to its schedule until it's resumed.
   - Click the resume icon - The report resumes on its previous schedule.

### Delete reports

To delete a report, follow these steps:

1. In the main menu, click **Dashboards > Reporting**.
1. On the row of the report you want to update, click the trash can icon.

> 1. Click **Delete** to confirm.

Author: ```suggestion
1. Click **Delete** to confirm.

Deleting a report is irreversible.
```

## Troubleshoot Reporting

To troubleshoot and get more log information, enable debug logging in the configuration file.
Refer to the [log filters configuration documentation](ref:log-filters) for more information.

```bash

> [log]

Reviewer 2: I can't write a suggestion because you didn't update it but the config should now be:
```
filters = rendering:debug,report.api:debug,report.render:debug,report.scheduler:debug,report.sender:debug,report.service:debug
```
```

1. When you've finished making changes, click **Confirm** at the top of the screen to go to the last step.
1. Click **Update report**.

### Pause or resume reports {#pause-a-report}

You can pause and resume sending reports from the report list view.
To do this, follow these steps:

1. In the main menu, click **Dashboards > Reporting**.
1. On the row of the report you want to update, do one of the following:

   - Click the pause icon - The report won't be sent according to its schedule until it's resumed.
   - Click the resume icon - The report resumes on its previous schedule.

### Delete reports

To delete a report, follow these steps:

1. In the main menu, click **Dashboards > Reporting**.
1. On the row of the report you want to update, click the trash can icon.
1. Click **Delete** to confirm.

## Troubleshoot Reporting

To troubleshoot and get more log information, enable debug logging in the configuration file.
Refer to the [log filters configuration documentation](ref:log-filters) for more information.

```bash
[log]
filters = report:debug
```

> 

Author: I assume this only applies to Enterprise. Is there a Cloud version of this?

1. When you've finished making changes, click **Confirm** at the top of the screen to go to the last step.
1. Click **Update report**.

### Pause or resume reports {#pause-a-report}

You can pause and resume sending reports from the report list view.
To do this, follow these steps:

1. In the main menu, click **Dashboards > Reporting**.
1. On the row of the report you want to update, do one of the following:

   - Click the pause icon - The report won't be sent according to its schedule until it's resumed.
   - Click the resume icon - The report resumes on its previous schedule.

### Delete reports

To delete a report, follow these steps:

1. In the main menu, click **Dashboards > Reporting**.
1. On the row of the report you want to update, click the trash can icon.
1. Click **Delete** to confirm.

## Troubleshoot Reporting

To troubleshoot and get more log information, enable debug logging in the configuration file.
Refer to the [log filters configuration documentation](ref:log-filters) for more information.

```bash
[log]
filters = report:debug
```

> <!-- is there a Cloud version of this? -->

Author: ```suggestion
```
